agent/consul/multilimiter/multilimiter.go

```diff
@@ -0,0 +1,128 @@
+package multilimiter
+
+import (
+	"context"
+	radix "github.com/hashicorp/go-immutable-radix"
+	"golang.org/x/time/rate"
+	"sync/atomic"
+	"time"
+)
+
+type LimitedEntity interface {
+	Key() []byte
+}
+
+type RateLimiter interface {
+	Start()
+	Stop()
+	Allow(entity LimitedEntity) bool
+	UpdateConfig(c Config)
+}
+
+type Limiter struct {
+	limiter    *rate.Limiter
+	lastAccess atomic.Int64
+}
+
+type Config struct {
+	Rate         rate.Limit
+	Burst        int
+	CleanupLimit time.Duration
+	CleanupTick  time.Duration
```

Comment:
Is this the the Interval which the routine checks to see if anything needs to be cleaned up?  If so, can we call it something like `CleanupCheckInterval`?